### ISSUE TITLE
Issues while creating NLB with type:LoadBalancer on port 80

### ISSUE THREAD

**Issue by allabalakrishna99** [kubernetes-sigs/aws-load-balancer-controller#3944]
**Describe the bug**
While creating NLB with type:LoadBalancer on port 80, when using aws-load-balancer-controller v2.9.2, getting below error when the same load balancer can be rolled out on port 443 without any issues

Failed deploy model due to operation error Elastic Load Balancing v2: DescribeListenerAttributes, https response error StatusCode: 403, RequestID: e476513e-2c4e-41ce-b1d4-7b967aaac6d0, api error AccessDenied: User: arn:aws:sts::690476720064:assumed-role/rafay-mgi-dev-eks-addon-iamserviceaccount-kub-Role1-CRVXhzK1iLmK/1731946561026787408 is not authorized to perform: elasticloadbalancing:DescribeListenerAttributes because no identity-based policy allows the elasticloadbalancing:DescribeListenerAttributes action

**Steps to reproduce**

Deploy below yaml file when using aws-load-balancer controller v2.9.2 on port 80:

apiVersion: v1
kind: Service
metadata:
  name: abc-service
  namespace: xyz
  annotations:
    service.beta.kubernetes.io/aws-load-balancer-type: "external"
    service.beta.kubernetes.io/aws-load-balancer-name: "abc"
    service.beta.kubernetes.io/aws-load-balancer-nlb-target-type: "ip"
    service.beta.kubernetes.io/aws-load-balancer-subnets: ''
    service.beta.kubernetes.io/aws-load-balancer-scheme: "internal"
    service.beta.kubernetes.io/load-balancer-source-ranges: ""
    service.beta.kubernetes.io/aws-load-balancer-backend-protocol: "tcp"
    service.beta.kubernetes.io/aws-load-balancer-healthcheck-port: "3049"
    service.beta.kubernetes.io/aws-load-balancer-cross-zone-load-balancing-enabled: "true"
spec:
  ports:
    - port: 80
      targetPort: 3049
      protocol: TCP
  type: LoadBalancer
  selector:
    app: abcxyz

**Expected outcome**
NLB Load balancer must be rolled out on port 80 

**Environment**

* AWS Load Balancer controller version : v2.9.2
* Kubernetes version: v1.30
* Using EKS (yes/no), if so version? Yes

**Additional Context**:

Reverting to lower version to v2.8 of aws-load-balancer-controller works

**Comment by zac-nixon** [comment]
Please apply the latest IAM policy: https://github.com/kubernetes-sigs/aws-load-balancer-controller/blob/a2e02479398b9f637c978db5081bb1775d65798b/docs/install/iam_policy.json#L44

**Comment by zac-nixon** [comment]
Is the problem that you see inconsistent behavior between port 80 vs port 443? The root cause is still that you haven't applied the updated IAM permissions.

The reason why you see this issue on port 80 is that listener attributes are currently not supported for TLS listeners, so we only attempt to modify listener attributes when the listener is non-TLS: https://github.com/kubernetes-sigs/aws-load-balancer-controller/blob/c701a42ce2e03c0dc18eea20449ef48d091192c8/pkg/deploy/elbv2/listener_manager.go#L381-L384.

TL;DR - Just update your IAM policy to use the new LBC version.

**Comment by zac-nixon** [comment]
I don't agree with making that behavior consistent.

**Comment by allabalakrishna99** [comment]
Can you please help me understand why that's the case? 

Why should rollout of a load balancer on port 443 allowed without adding the required permissions which is not allowed port 80?

**Comment by zac-nixon** [comment]
I explained it above:

The reason why you see this issue on port 80 is that listener attributes are currently not supported for TLS listeners, so we only attempt to modify listener attributes when the listener is non-TLS: https://github.com/kubernetes-sigs/aws-load-balancer-controller/blob/c701a42ce2e03c0dc18eea20449ef48d091192c8/pkg/deploy/elbv2/listener_manager.go#L381-L384.
